contrib/fparser/fparser_ad.cc

```diff
@@ -677,18 +679,23 @@ bool FunctionParserADBase<Value_t>::JITCompileHelper(const std::string & Value_t
 
   // tmp filenames
   char ccname[] = "./tmp_jit_XXXXXX";
-  if (mkstemp(ccname) == -1)
+  int ccfile = mkstemp(ccname);
+  if (ccfile == -1)
   {
     std::cerr << "Error creating JIT tmp file " << ccname << ".\n";
     return false;
   }
-  char object[] = "./tmp_jit_XXXXXX";
-  if (mkstemp(object) == -1)
+  close(ccfile); // close file. We reopen this using an ofstream below.
+
+  char objectname[] = "./tmp_jit_XXXXXX";
+  int objectfile = mkstemp(objectname);
+  if (objectfile == -1)
   {
-    std::cerr << "Error creating JIT tmp file " << object << ".\n";
+    std::cerr << "Error creating JIT tmp file " << objectname << ".\n";
     std::remove(ccname);
     return false;
   }
+  close(objectfile); // close file. Will be reopened by the compiler.
```

Comment:
I guess we should probably just be doing `fprintf()`/C-based I/O for the files if we find there is a performance hit....